graphs/fit_atg_model.py

```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+from typing import List
+
+import numpy as np
+import pytest
+from scipy.optimize import least_squares
+
+# TODO(miskosz): Pass in as a parameter?
+EXPONENT = 6.23
+
+@dataclass
+class AtgModelFit:
+    """
+    Result of fitting (x, y) data samples to the curve:
+        y = (a/tg) * (x/tg)^6.23 * e^(-x/tg)
+    
+    TODO(miskosz): Maybe return also something like a standard deviation of the fit.
+    """
+    a: float
+    tg: float
+
+    def predict(self, x: float) -> float:
+        ys = _model(params=[self.a, self.tg], xs=[x])
+        return ys[0]
+
+
+def fit_atg_model(xs: np.ndarray, ys: np.ndarray) -> AtgModelFit:
+    """
+    Fits atg model through `(xs, ys)` datapoints.
+    """
+    assert len(xs) == len(ys), "Inconsistent number of datapoints to fit."
+    assert np.all(xs > 0), "No support for non-positive values for `xs`."
+    assert np.all(ys >= 0), "No support for negative values for `ys`."
+    a0 = 2000.0
+    tg0 = 7.0
+    # TODO(miskosz): Handle failures.
+    least_squares_result = least_squares(fun=_residuals, x0=[a0, tg0], args=(xs, ys))
+    return AtgModelFit(a=least_squares_result.x[0], tg=least_squares_result.x[1])
+
+
+def _residuals(params: List[float], xs: np.ndarray, ys: np.ndarray) -> float:
+    """
+    Returns the residual ("error") of model fitting with parameter values `params`
+    to the datapoints (xs, ys).
+
+    NOTE: We compute the logarithm in logspace since the data spans multiple orders.
+    We don't care if the model is off by 10 for values close to 20000, but we care
+    for small values. We choose natural base of the logarithm.
+    """
+    return np.log(1. + _model(params=params, xs=xs)) - np.log(1. + ys)
+
+
+def _model(params: List[float], xs: np.ndarray) -> np.ndarray:
+    """
+    Returns predicted y-values of the model
```

Comment:
Môžeš túto funkciu použiť tu? https://github.com/lukipuki/COVID-19-simulation/blob/3c70ab6011c5745944eea0ac3f0f4ab1e2ae7550/graphs/country.py#L37

Reply:
Hej, zjednotim to nejak.. in an upcoming PR. Potrebujem totiz oddelit nacitavanie a prezentaciu dat v `country.py`.

Reply:
:+1: